lib/YaST/Module.pm

```diff
@@ -47,7 +47,7 @@ sub open {
     else {
         die "Unknown user interface: $ui";
     }
-    YuiRestClient::get_app()->check_connection();
+    YuiRestClient::get_app()->check_connection(timeout => $args{timeout});
```

Comment:
(optional) it is a good habit to unpack all the arguments at the begining of the subroutine (before the first `die` in this code. To keep some order helps in Perl as there are not built-in signatures.